pkg/metrics/providers/keptn.go

```
@@ -0,0 +1,252 @@
+		return 0, fmt.Errorf("could not retrieve KeptnMetric %s/%s: %w", queryObj.Namespace, queryObj.ResourceName, err)
+	}
+
+	if status, ok := get.Object["status"]; ok {
+		if statusObj, ok := status.(map[string]interface{}); ok {
+			if value, ok := statusObj["value"].(string); ok {
+				floatValue, err := strconv.ParseFloat(value, 64)
+				if err != nil {
+					return 0, fmt.Errorf("could not parse value of KeptnMetric %s/%s to float: %w", queryObj.Namespace, queryObj.ResourceName, err)
+				}
+				return floatValue, nil
+			}
+		}
+	}
+	return 0, fmt.Errorf("could not retrieve KeptnMetric - no value found int resource %s/%s", queryObj.Namespace, queryObj.ResourceName)
+}
+
+func (k *KeptnProvider) queryKeptnAnalysis(obj *queryObject) (float64, error) {
+	analysis := &unstructured.Unstructured{
+		Object: map[string]interface{}{
+			"apiVersion": fmt.Sprintf("metrics.keptn.sh/%s", apiVersion),
+			"kind":       "Analysis",
+			"metadata": map[string]interface{}{
+				"name":      fmt.Sprintf("%s-%s", obj.ResourceName, uuid.New().String()[:6]),
+				"namespace": obj.Namespace,
+			},
+			"spec": map[string]interface{}{
+				"analysisDefinition": map[string]interface{}{
+					"name": obj.ResourceName,
+				},
+				"timeframe": map[string]interface{}{
+					"recent": obj.DurationString,
+				},
+				"args": obj.Arguments,
+			},
+		},
+	}
+
+	// set the timeout to 10s - this will give Keptn enough time to reconcile the Analysis
+	// and store the result in the status of the resource created here.
+	ctx, cancel := context.WithTimeout(context.Background(), k.analysisTimeout)
+	defer cancel()
+
+	createdAnalysis, err := k.client.
+		Resource(obj.GroupVersionResource).
+		Namespace(obj.Namespace).
+		Create(ctx, analysis, v1.CreateOptions{})
+
+	if err != nil {
+		return 0, fmt.Errorf("could not create Keptn Analysis %s/%s: %w", obj.Namespace, obj.ResourceName, err)
+	}
+
+	// delete the created analysis at the end of the function
+	defer func() {
+		_ = k.client.
```

instead of ignoring the error, we should log it.